lib/spack/spack/spec.py

```diff
@@ -426,7 +447,56 @@ def __init__(self, spec_like, *dep_like, **kwargs):
         # Spec(a, b) will copy a but just add b as a dep.
         for dep in dep_like:
             spec = dep if isinstance(dep, Spec) else Spec(dep)
-            self._add_dependency(spec)
+            self._add_dependency(spec, (spack.DEP_BUILD, spack.DEP_INSTALL))
```

Comment:
Rather than assuming defaults here (and writing the default in yet another place) I would leave this empty to start with.

Rationale: When you read in a spec from a literal, you've got no way of knowing what dep types there are.  For all the other fields in `Spec`, the convention is that we leave them empty or `None` until either `concretize()` or `normalize()` fills them in.  Here, I would say to fill this in based on `normalize()`, because `normalize()` is supposed to merge information (like constraints) from `package.py` files with what is in the user-provided `Spec`  Since the current `package.py` tells you what the deptypes are, `normalize` seems like the right place to assign this.